### ISSUE TITLE
In the archived room, the user has access to the IOU editing panel in the RHP

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#36371]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.40-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:**

## Action Performed:
1. Open https://staging.new.expensify.com/
2. Log in under your HT account
3. Create WS
4. Send the IOU to the WS room
5. Delete the created WS
6. Navigate to the archived WS room with the sent IOU
7. Open the details of the IOU
8. Try to edit any IOU details

## Expected Result:
If the IOU is in the archive room, the user should not be able to edit details. The IOU should look like a paid IOU but without the green check mark.

## Actual Result:
In the archived room, the user has access to the IOU editing panel in the RHP.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/f9377285-987f-4153-b2b5-2ade95308c8e

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by m-natarajan** [comment]
We (Applause) think this might be related to #vip-bills
cc @davidcardoza

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?

The root cause of the problem is that `canEditMoneyRequest` and `canEditFieldOfMoneyRequest` functions do not check if the report has been archived.

https://github.com/Expensify/App/blob/59559c446dccaea53a30f830c3a2c67f5b52ddbb/src/libs/ReportUtils.ts#L2060-L2062

https://github.com/Expensify/App/blob/59559c446dccaea53a30f830c3a2c67f5b52ddbb/src/libs/ReportUtils.ts#L2073

### What changes do you think we should make in order to solve the problem?


We should check if the report has been archived over [here](https://github.com/Expensify/App/blob/59559c446dccaea53a30f830c3a2c67f5b52ddbb/src/libs/ReportUtils.ts#L2060-L2062) and [here](https://github.com/Expensify/App/blob/59559c446dccaea53a30f830c3a2c67f5b52ddbb/src/libs/ReportUtils.ts#L2073) to improve the function `canEditMoneyRequest`.

Therefore, we will simply use the `isArchivedRoom` function to return false when it's true.

### What alternative solutions did you explore? (Optional)

However, we can use the `isArchivedRoom` function at the `canEditFieldOfMoneyRequest` function if the field is amount.

**Comment by Tony-MK** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?

The root cause of the problem is that `canEditMoneyRequest` and `canEditFieldOfMoneyRequest` functions do not check if the report has been archived.

https://github.com/Expensify/App/blob/59559c446dccaea53a30f830c3a2c67f5b52ddbb/src/libs/ReportUtils.ts#L2060-L2062

https://github.com/Expensify/App/blob/59559c446dccaea53a30f830c3a2c67f5b52ddbb/src/libs/ReportUtils.ts#L2073

### What changes do you think we should make in order to solve the problem?


We should check if the report has been archived over [here](https://github.com/Expensify/App/blob/59559c446dccaea53a30f830c3a2c67f5b52ddbb/src/libs/ReportUtils.ts#L2060-L2062) and [here](https://github.com/Expensify/App/blob/59559c446dccaea53a30f830c3a2c67f5b52ddbb/src/libs/ReportUtils.ts#L2073) to improve the function `canEditMoneyRequest`.

Therefore, we will simply use the `isArchivedRoom` function to return false when it's true.

### What alternative solutions did you explore? (Optional)

However, we can use the `isArchivedRoom` function at the `canEditFieldOfMoneyRequest` function if the field is amount.

https://github.com/Expensify/App/blob/59559c446dccaea53a30f830c3a2c67f5b52ddbb/src/libs/ReportUtils.ts#L2108-L2110

**Comment by situchan** [comment]
reviewing proposal

**Comment by jacobkim9881** [comment]
Should this cared as an issue? Since the chat room is archived, a user can't do anything without paid button and can see it is archived as a room name.

**Comment by situchan** [comment]
IOU reports won't be archived after deleting workspace

**Comment by Tony-MK** [comment]
Hi @situchan, thanks for reviewing my proposal.

After some investigation, I have a quick question.

I think the IOU reports do get archived because when trying to edit the amount field, the following response is received:

![Screenshot 2024-02-21 at 00 27 10](https://github.com/Expensify/App/assets/22982173/6ec75e2c-42ad-452c-942a-a28a0bcf272d)

Therefore, is it reasonable to check if the workspace is archived and deny the user the ability to edit the amount field?

The other IOU fields show the `NotFoundPage` component. 

Should we mimic this behavior instead?

Thank you.

**Comment by jacobkim9881** [comment]
@Tony-MK question seems reasonable. It seems editing amount returns error and editing other field is blocked too? These are different to Actual Result above.

**Comment by situchan** [comment]
@Tony-MK please test on latest version. IOU reports don't get archived after deleting workspace

**Comment by Tony-MK** [comment]
@situchan Thanks for help.

I updated my proposal based on the comment ⬆️.

Could you review it?

However, I also noticed that the behavior I mentioned in https://github.com/Expensify/App/issues/36371#issuecomment-1955160497 also occurs.

**Comment by situchan** [comment]
still discussing proposals

**Comment by situchan** [comment]
Not sure if something changed recently. Now all money request reports are archived after deleting workspace.
This should not happen as per https://github.com/Expensify/App/issues/35174#issuecomment-1926537796
@cristipaval can you please confirm?

**Comment by Tony-MK** [comment]
> Not sure if something changed recently. Now, all money request reports are archived after deleting workspace.

Hi @situchan, the backend archives the report, not the front end, because while deleting the workspace offline, the IOU report is not archived until the user goes online.

The server side changes discussed in this https://github.com/Expensify/App/issues/29049#issuecomment-1767496708 and decided in its https://github.com/Expensify/App/issues/29049#issuecomment-1822616337 were made to archive task threads also archived IOUs report.

#

The screenshot below shows the IOU report, which has a `key` of `report_7758574277954490` from the part of the `command=DeleteWorkspace` API response.

Note the `statusNum` and `stateNum` are set to `2` to mean it's closed.

![Network Logs](https://github.com/Expensify/App/assets/22982173/6923a1fd-85bf-48f8-a56b-3848fc4629ac)

**Comment by situchan** [comment]
Waiting for confirmation [here](https://github.com/Expensify/App/issues/36371#issuecomment-1968976799)

**Comment by cristipaval** [comment]
I asked internally [here](https://expensify.slack.com/archives/C03U7DCU4/p1709650968265099)

**Comment by cristipaval** [comment]
Assigned myself for now until the internal discussion ends

**Comment by cristipaval** [comment]
> Not sure if something changed recently. Now all money request reports are archived after deleting workspace. This should not happen as per [#35174 (comment)](https://github.com/Expensify/App/issues/35174#issuecomment-1926537796) @cristipaval can you please confirm?

Yes, the money requests are now archived and we might change this soon. But until we change this, I think we should proceed with the fix for the current issue and block the editing.

**Comment by stephanieelliott** [comment]
Ok so based on the discussion it sounds like we should update the front end to match the future back end implementation plans and not allow editing on an archived transaction. Now that that's clarified, I think we can proceed with the existing proposals, right @situchan? Are there any proposals here that will work?

**Comment by stephanieelliott** [comment]
Another internal conversation about this here: https://expensify.slack.com/archives/C036QM0SLJK/p1710283476740279

**Comment by cristipaval** [comment]
I'm closing this one based on where we landed in the conversation liked by @stephanieelliott above

**Comment by Tony-MK** [comment]
> I'm closing this one based on where we landed in the conversation linked by @stephanieelliott above

I don't know what occurred in the [slack conversation](https://expensify.slack.com/archives/C036QM0SLJK/p1710283476740279) but unpaid money requests get archived only backend.

Since the front end doesn't get archive money requests offline, shouldn't we address this to adhere to our patterns?

Stephanie mentioned it in this https://github.com/Expensify/App/issues/36371#issuecomment-1992692042 and I am curious. 🤔 

Thanks.
